### ISSUE TITLE
[False Positive] Obsidian `Component` is identified as `Promise`

### ISSUE THREAD

**Issue by hesprs** [obsidianmd/eslint-plugin#181]
This false positive is so obvious and hilarious. The root cause is in Obsidian `BaseComponent` definition, `BaseComponent` has method `then`:

```TypeScript
export abstract class BaseComponent {
    /**
     * Facilitates chaining
     * @public
     * @since 0.9.7
     */
    then(cb: (component: this) => any): this;
}
```

The linter regards a component is thenable, then flags false positive `typescript/no-misused-promises` to any conditional statements targeting any Obsidian component (since they all extend `BaseComponent`). Minimal reproduction:

```TypeScript
import type { BaseComponent } from 'obsidian';

declare const component: BaseComponent;
if (component) console.log('False positive!');
```

`typescript/no-misused-promises` should be falsely triggered at `if (component) ...`.

**Comment by saberzero1** [comment]
It seems this would require changes to an upstream dependency. I'll raise an issue.

**Comment by saberzero1** [comment]
https://github.com/typescript-eslint/typescript-eslint/issues/12673

It seems the recent PRs sent out to prepare theme authors for the 1.13 breaking changes to `--callout-color`/`--canvas-color` flag my account for their automated tooling.
